### ISSUE TITLE
[Telemetry] Credit Card Autofill 

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#18711]
## Description & Product Manager / Data Scientist User Story

As a product manager, I need to be able to measure the success of the new Credit Card Autofill feature, and understand pain points that we can fix in the next iterations. 

## What questions will you answer with this data?

See success criteria here: https://docs.google.com/document/d/1MDjCb2zFXrDqdFZOAqSGQez4Y0CDhnVZkzTfhmsryBM/edit#

**Note: ** save from autofill has been descoped for now. 

## Acceptance Criteria
- [x] ENG files a [DS JIRA](https://jira.mozilla.com/projects/DO/issues/DO-228?filter=allopenissues) request outlining their methodology.
- [x] DS sign off on instrumentation methodology addressing product questions.
- [ ] Event pings can be queried via re:dash
- [ ] Event pings can be queried via amplitude
- [x] We are sending telemetry events for the actions listed in the requirements
- [x] We have documented the telemetry
- [x] We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry

**Comment by Mugurell** [comment]
@gabrielluong With the feature not being complete do you think this is actionable at this time or it should be labeled as `waiting` for until we have the final functionality in place?

**Comment by gabrielluong** [comment]
> @gabrielluong With the feature not being complete do you think this is actionable at this time or it should be labeled as `waiting` for until we have the final functionality in place?

We landed the select credit card prompt. We descoped the save/update prompts from 90 since that was blocked on GV. In every sense, we are feature complete for 90, but with a lot of polish work to be done. I think whoever takes this on should work with @vesta0 and @kbrosnan knowing that somethings are descoped from the requirements.

**Comment by eliserichards** [comment]
Data science ticket filed: https://jira.mozilla.com/browse/DS-1598

**Comment by eliserichards** [comment]
Autofill save and manual save should be counted separately (confirmed with Channing)

**Comment by gabrielluong** [comment]
For QA testing:

- We want to verify that we get `credit_cards.autofill_card` count  in the metrics ping when a credit card is autocompleted into a credit form. Related to https://github.com/mozilla-mobile/fenix/issues/18287. 
- A count in the metrics ping when we delete a credit card `credit_cards.delete_card`
- A count in the metrics ping when we add a credit card `credit_cards.manual_save`

**Comment by abodea** [comment]
Hello @gabrielluong, note that I tested this on your PR from 9h ago with a debug build.
I was able to generate the metrics ping but I couldn't find any of `credit_cards.manual_save`, `credit_cards.delete_card` or  `credit_cards.autofill_card`.
Please check the following for more reference: [metrics ping](https://debug-ping-preview.firebaseapp.com/pings/Abodea4), [metrics ping](https://debug-ping-preview.firebaseapp.com/pings/Abodea3).

Removing `eng:qa:needed` until further notice.

**Comment by gabrielluong** [comment]
It was probably landed 1 hour before you tested it. So, I imagine we just need to test with latest build.

**Comment by abodea** [comment]
> It was probably landed 1 hour before you tested it. So, I imagine we just need to test with latest build.

@gabrielluong please note that I tested it exactly on your PR.

**Comment by gabrielluong** [comment]
@abodea Added a fix in https://github.com/mozilla-mobile/fenix/pull/19733. 

For QA testing:

- We want to verify that we get `metrics.credit_cards_autofill_count` count  in the metrics ping when a credit card is autocompleted into a credit form. Related to https://github.com/mozilla-mobile/fenix/issues/18287. 
- A count in the metrics ping when we delete a credit card `credit_cards_deleted_count `
- A count in the metrics ping when we add a credit card `credit_cards_saved_count `

**Comment by lobontiumira** [comment]
Verified as fixed on the Beta 90.0.0-beta.1 build with Google Pixel (Android 10).
The metrics were generated for the:
* credit_cards_autofill_count
* credit_cards_deleted_count
* credit_cards_saved_count

Ping: 74facfac-4c13-44bb-9b7c-3773e7bd5b5f
[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/mira-2106020006)
[logcat.txt](https://github.com/mozilla-mobile/fenix/files/6584346/logcat.txt.txt)

Note: also verified on the 6/2 Nightly build, and the credit_cards metrics generated as mentioned above.
